### PR TITLE
Guarding backbone page transitions

### PR DESCRIPTION
**Pull request by kjcpaas**
Qlearn has a page guard transition feature that works even on Backbone. (@banyan started it for Tutor, and I improved for assessment) I'd like to share how it's implemented in our tech blog to help other developers.

Relevant code snippets are:

https://github.com/quipper/qlearn-web/tree/develop/src/coffee/modules/backbone_transitions/backbone_transitions.coffee
https://github.com/quipper/qlearn-web/blob/develop/src/coffee/views/app.coffee#L89L91

### CODE REVIEW

**Comment by kjcpaas** [comment]
@quipper/developers Please read and review. :smile_cat:

File: _posts/2015-07-31-guarding-backbone-page-transitions.markdown [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+One thing to keep in mind for handlers is that if it returns `null`, it will not guard the page transition (popup won't be shown). On the other hand, if a string is returned, this string will be shown in the popup.
+
+![](http://i.gyazo.com/4b097d47b068b013a79ec4ceda0d6d83.png)
+
+Handlers can be attached using the following code:
+
+{% highlight coffeescript %}
+guardPage = (e) -> 'Your new changes will be lost.'
+dontGuardPage = (e) -> nul
```
Comment (banyan) [inline _posts/2015-07-31-guarding-backbone-page-transitions.markdown]: typo? `nul`

Comment (kjcpaas) [inline _posts/2015-07-31-guarding-backbone-page-transitions.markdown]: Ah yes. Thanks for the catch! :smile:

**Comment by sethjeffery** [comment]
Cool! :sunglasses: :+1:

**Comment by kjcpaas** [comment]
Publishing this now :smile_cat:

**Pull request merged**